core/web/loader/job_spec_errors.go

```diff
@@ -0,0 +1,60 @@
+package loader
+
+import (
+	"context"
+
+	"github.com/graph-gophers/dataloader"
+
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/utils/stringutils"
+)
+
+type jobSpecErrorsBatcher struct {
+	app chainlink.Application
+}
+
+func (b *jobSpecErrorsBatcher) loadByJobIDs(_ context.Context, keys dataloader.Keys) []*dataloader.Result {
+	// Create a map for remembering the order of keys passed in
+	keyOrder := make(map[string]int, len(keys))
+	// Collect the keys to search for
+	var jobIDs []int32
+	for ix, key := range keys {
+		id, err := stringutils.ToInt32(key.String())
+		if err == nil {
+			jobIDs = append(jobIDs, id)
+		}
+
+		keyOrder[key.String()] = ix
+	}
```

Comment:
I notice we do the same song and dance for a lot of these batchers.  Seems like a good candidate for combing all of them into a more abstract batcher type using generics?

Just noting this as a suggestion for a future project, no need to include it in this PR.

Reply:
Yes I think there could be a few generics-based loaders that would be handy

Reply:
I will create a story to track this generics work around loaders.